feat: use passport-jwt for authentication

sea-gull-diana opened the pull request:
## What does this PR do ?
Modifies user authentication by using passport-jwt strategy.

### TODO
Add refresh tokens and possibly API tokens (for CLI login).

sea-gull-diana committed:
feat: use passport-jwt for authentication

BioTheWolff commented on src/auth/auth.service.ts:
```diff
@@ -28,17 +28,31 @@ export class AuthService {
     // Add a username in the JWT token
-    const payload = {username: user.username, isAdmin: user.isAdmin};
+    const payload = {
+      sub: user['id'],
```
User ID should not be disclosed. Should we use username, or change the schema to use UUIDs even for users?

sea-gull-diana replied:
It's not exactly disclosing since we are putting this payload into a JWT token, no? Anyways, I do prefer the user id to be a uuid, it seems like a clean way to do things.

BioTheWolff commented on src/auth/auth.service.ts:
```diff
@@ -28,17 +28,31 @@ export class AuthService {
+    const user = await this.usersService.findOneUnsanitized({email: credentials.email});
+    if (user && await bcrypt.compare(credentials.password, user.password)) {
+      return {
+        id: user.id,
```
Same goes for here, only the username is important

Commit:
Database refactor: uuids and uniques (#11)

Refactor the database by changing all integer IDs into UUIDs, except for
the SSH keys, who benefit from staying in "normal" IDs; Remove the
unique constraints where they shouldn't have been, in order to avoid
problems; Add password nonce to enable generation of a "forgotten
password" email.

BREAKING CHANGE

sea-gull-diana committed:
fix: remove unused uuid imports, search/delete project by id instead of uuid

BioTheWolff reviewed (approved):
LGTM, thanks for the changes!

mfernd reviewed (approved):
lgtm
